### PR TITLE
refactor: explicit `plot_options` & use `super().plot(...)`

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

This resolves typing issue as mypy can't trace the dict exxiciently, and also we have lots duplicated code so lets have it not implemented for remaining

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by SkafteNicki** [comment]
PR looks good to me, but after this PR https://github.com/Lightning-AI/metrics/pull/1485 was merged a few metrics are missing to be updated.

**Comment by Borda** [comment]
> PR looks good to me, but after this PR #1485 was merged a few metrics are missing to be updated.

I do not see any `plot_option` as I did this last night so it shall be very aligned with latest set of master :chipmunk:

**Comment by SkafteNicki** [comment]
> > PR looks good to me, but after this PR #1485 was merged a few metrics are missing to be updated.
> 
> I do not see any `plot_option` as I did this last night so it shall be very aligned with latest set of master 🐿️

Here is an example: https://github.com/Lightning-AI/metrics/blob/df15e09e1a1581a737e0ef31b87adfbf84a1877b/src/torchmetrics/aggregation.py#L160-L201

**Comment by Borda** [comment]
> Here is an example:

ohh, I see it is different pattern, will add it :)

**Review by SkafteNicki — APPROVED** [review]

File: src/torchmetrics/metric.py [hunk 1]
```diff
@@ -569,6 +573,33 @@ def plot(self, *_: Any, **__: Any) -> Any:
...
+            ModuleNotFoundError:
+                If `matplotlib` is not installed
+        """
+        val = val or self.compute()
+        fig, ax = plot_single_or_multi_val(
+            val,
+            ax=ax,
+            higher_is_better=self.higher_is_better,
+            name=self.__class__.__name__,
+            lower_bound=self._plot_lower_bound,
```
Comment (justusschock) [inline src/torchmetrics/metric.py:597]: can we make these not protected but public? They are kind of public API similar to `higher_is_better`

**Review by justusschock — APPROVED** [review]

File: src/torchmetrics/metric.py (hunk 1 above)
Comment (Borda) [reply src/torchmetrics/metric.py:597]: personaly I would keep this and protected so user shall not edit it... well we can ban these from editing, then set it public :)

Comment (Borda) [reply src/torchmetrics/metric.py]: check, mind have look :)

**Pull request merged**